common/workunit/workunit.cpp

```diff
@@ -2487,6 +2487,39 @@ class CompoundAggregator : implements StatisticAggregator
     IArrayOf<StatisticAggregator> aggregators;
 };
 
+class CostAggregator : public StatisticAggregator
+{
+public:
+    CostAggregator(StatsAggregation & _summary) : StatisticAggregator(StCostExecute), summary(_summary) {}
+
+    virtual void noteStatistic(StatisticKind kind, unsigned __int64 value, IConstWUStatistic & extra) override
+    {
+        summary.noteValue(value);
+    }
+
+protected:
+    StatsAggregation & summary;
+};
+
+extern WORKUNIT_API __int64 aggregateCost(IConstWorkUnit * wu)
+{
+    const WuScopeFilter filter("stat[CostExecute],depth[0..3],nested[0]");
```

Comment:
am not quite sure of the depth[0..3] semantics here.
Perhaps needs a comment added here to explain.

Reply:
Do you actually want to match global costs?  If so it will double count them.  I suspect the depth should be [1..3]